grunttasks/scan-email-translations.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  grunt.registerTask('scan-email-translations', 'Scan the generated emails for invalid style tags', [
+    'generate-valid-styles',
+    'verify-styles'
+    ]
+  );
+
+  grunt.registerMultiTask('verify-styles', 'Verify whether the translated styles are valid', function (){
+    var fileArray = this.files[0].src;
+    var contents = grunt.file.read(fileArray.shift());
+    var styleArray = JSON.parse(contents);
+
+    var hasErrors = 0;
+    var styleRegex = /style='(.*)'/ig;
```

Comment:
@TDA I think we can make it non-greedy by changing the RegExp to something like this:
```js
var styleRegex = /style='(.*?)'/ig;
```

That way it will stop on the first `'` it encounters. But yeah, maybe we need to rethink this and do what @rfk suggests, which is comparing the entire HTML tag's contents. Because I can break the above RegExp by using double quotes instead of single quotes, and turns out there's an even bigger edge case of multiline tags.

For example, I was curious why I could completely edit the `style` attribute in the French locale and it wouldn't detect it, and then after adding a few `console.log()` in the Grunt files, I think it is because the `style=''` starts on one line and closes on another line (which seems to explain why in some cases `styleList` had 1 result and in other cases had 2 -- where the 1 result was probably where it matches the English key in the .po files):

```pot
#: templates/verify.html:7
msgid ""
"For more information, please visit <a href='https://support.mozilla.org' "
"style='color: #0095dd; text-decoration: none;' target='nein'>Mozilla Support</a>"
msgstr ""
"Pour davantage d'informations, veuillez consulter <a "
"href='https://support.mozilla.org' style='color: green; display: none; text-decoration: "
"none;'>l'assistance de Mozilla</a>"
```

We'd have to some more research, but we may need to do something like a [**grunt-po2json**](https://github.com/rockykitamura/grunt-po2json) where we convert the .po files to .json and then do the scanning.